x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/In.java

```diff
@@ -84,11 +84,12 @@ public boolean foldable() {
 
     @Override
     public Boolean fold() {
+        // Optimization for early return and Query folding to LocalExec
         if (value.dataType() == DataType.NULL) {
             return null;
         }
         if (list.size() == 1 && list.get(0).dataType() == DataType.NULL) {
```

Comment:
What does this line do?

Reply:
it's the same optimization to fold to LocalExec. I can merge the ifs